### PR TITLE
doc: add documentaion for rbd-nbd mounter

### LINKED ISSUE #2124: Qualify data consistent snapshots and cloning with NBD backend
# Describe the feature you'd like to have #

rbd-nbd backends support quiesce feature. While taking the snapshots rbd will notify rbd-nbd to call the quiesce hooks, upon receiving the notification the rbd-nbd will freeze the mount points and wait to finish the inflight IO. Once the snapshot is completed, rbd will notify the unquiesce, now rbd-nbd will unfreeze the mount points and allow IO to continue.

# What is the value to the end user? (why is it a priority?) #

Data consistent snapshots and clones.

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #

Let's have a separate doc for rbd-nbd.

## Related issues ##

Closes #2124

Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

### CODE REVIEW

File: docs/rbd-nbd.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Status
+
+Rbd-nbd support status: **Alpha**
+
+## Support Matrix
+
+| Features                                 | Feature Status | CSI Driver Version | Ceph Cluster Version | CSI Spec Version | Kubernetes Version |
+| ---------------------------------------- | -------------- | ------------------ | -------------------- | ---------------- | ------------------ |
+| Creating and deleting snapshot           | Alpha          | >= v3.4.0          | Pacific (>=16.0.0)   | >= v1.0.0        | >= v1.17.0         |
+| Creating and deleting clones             | Alpha          | >= v3.4.0          | Pacific (>=16.0.0)   | >= v1.0.0        | >= v1.17.0         |
```
Comment (Madhu-1) [inline docs/rbd-nbd.md:40]: what bug fixes we have done on this area(snapshot and clone) and why ceph cluster version should be 16+?

Comment (pkalever) [reply docs/rbd-nbd.md:40]: Madhu, Pacific has the rbd-nbd reattach capability, which will help attach the previously mapped devices to a new process (which we use after plugin restart)

File: docs/rbd-nbd.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+client-side, which is inside the `csi-rbdplugin` node plugin.
+
+## Configuration
+
+To use the rbd-nbd mounter for RBD-backed PVs, set `mounter` to `rbd-nbd`
+in the StorageClass.
+
+## Status
+
+Rbd-nbd support status: **Alpha**
```
Comment (nixpanic) [inline docs/rbd-nbd.md:33]: this should probably be linked from the main [support matrix](https://github.com/ceph/ceph-csi/blob/devel/README.md#support-matrix).

Comment (pkalever) [reply docs/rbd-nbd.md:33]: Good idea, I will refresh.

**Review by Madhu-1 — DISMISSED** [review]

File: docs/rbd-nbd.md (hunk 2 above)
Comment (Madhu-1) [reply docs/rbd-nbd.md:33]: Need to mention other operations? like CreateVolume ,expand etc?

Comment (pkalever) [reply docs/rbd-nbd.md:33]: We are not done on this Madhu, as you there are issues opened for qualifying encryption, expansion, thick provisioning, mirroring for rbd-nbd. With each of these issues, we are going to work closely with this doc again.

But, if you want Create Volume, Delete Volume to be added with this PR, yes that also make sense too, but just saying we are not done with this doc yet. I will leave it to you, please feel free to let me know your further thought on this.

Thanks!

Comment (Madhu-1) [reply docs/rbd-nbd.md:33]: Am fine if we are adding it later also

Comment (pkalever) [reply docs/rbd-nbd.md:33]: Cool. Sure @Madhu-1, we also have one issue for verifying all access modes and volume modes for rbd-nbd, may be we can test on access modes and volumes modes and then add support matrix accordingly for volume provisioning too.

File: README.md [hunk 3]
```diff
@@ -59,6 +59,9 @@ NOTE:
 
 ### Ceph-CSI features and available versions
 
+Please refer [rbd nbd mounter](https://github.com/ceph/ceph-csi/blob/devel/docs/rbd-nbd.md#support-matrix)
```
Comment (Madhu-1) [inline README.md]: Please add the relative path or else if someone opens this on another branch (release branch) this will point to the devel branch doc.

Comment (pkalever) [reply README.md]: Sure Madhu, I saw other docs using the complete path, so thought this is common here.

Comment (Madhu-1) [reply README.md]: sorry for that one. that can be correct later. but i hope you got the problem

Comment (pkalever) [reply README.md]: Oh yes! Thanks! Done now.

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**